src/fragmentarium/ui/SearchForm.test.tsx

```diff
@@ -53,34 +56,27 @@ async function renderSearchForms() {
 
 beforeEach(async () => {
   searchEntry = await factory.build('bibliographyEntry')
-  fragmentService = {
-    searchBibliography: jest.fn(),
-  }
-  fragmentService.searchBibliography.mockReturnValue(
-    Promise.resolve([searchEntry])
-  )
-  fragmentSearchService = {
-    random: jest.fn(),
-    interesting: jest.fn(),
-  }
-  session = {
-    isAllowedToReadFragments: jest.fn(),
-    isAllowedToTransliterateFragments: jest.fn(),
-  }
-  session.isAllowedToReadFragments.mockReturnValue(true)
-  session.isAllowedToTransliterateFragments.mockReturnValue(true)
+  fragmentService.searchBibliography = jest
+    .fn()
+    .mockReturnValue(Promise.resolve([searchEntry]))
+
+  fragmentSearchService.random = jest.fn()
+  fragmentSearchService.interesting = jest.fn()
+
+  session.isAllowedToReadFragments = jest.fn().mockReturnValue(true)
+  session.isAllowedToTransliterateFragments = jest.fn().mockReturnValue(true)
```

Comment:
Do not mutate services, especially when they are global. Class mocks already have all methods spied. Otherwise use `spyOn`.

Reply:
If `session` is an automatic mock, you should be able to just do `session.isAllowedToReadFragments..mockReturnValue(true)`. The mock constructor has to type `jest.Mock<jest.Mocked<MemorySession>>`.

If it is a normal class, you can spy it `jest.spyOn(session, 'isAllowedToReadFragments')` and then set the return value.
See:  https://jestjs.io/docs/en/jest-object#jestspyonobject-methodname